src/clang_parser.cpp

```diff
@@ -166,8 +166,10 @@ static SizedType get_sized_type(CXType clang_type)
 void ClangParser::parse(ast::Program *program, BPFtrace &bpftrace, std::vector<std::string> extra_flags)
 {
   auto input = program->c_definitions;
-  if (input.size() == 0)
-    return; // We occasionally get crashes in libclang otherwise
+  if (input.size() == 0) {
+    // We occasionally get crashes in libclang otherwise
+    input = "int main() { return 0; }";
```

Comment:
Why this? I think it could make the libclang crashes more common on static builds

Reply:
If we include files with our `--include` flag, but we don't have a `c_definitions` section in our bpftrace script, clang parser will skip and the user won't get the definitions from the file they tried to include.

I guess we could also set input as `#define __BPFTRACE_DUMMY__` or something, the only requirement is that we don't skip the clang parsing stage if the user don't have any C definitions in their script.

Why would it crash more frequently on static builds?

Reply:
Most scripts don't actually need any Clang stuff, so avoiding running it here when it's not required makes the static build usable mostly

Reply:
But how to detect scripts that need clang stuff?

Reply:
Adding something like `#define __BPFTRACE_DUMMY__ ` to the input in main.cpp when `include_files` is not empty should work

Reply:
Ok, makes sense, I'll update it tomorrow